src/publisher.rs

```diff
@@ -239,11 +240,13 @@ pub unsafe extern "C" fn z_publisher_put(
         }
     }
 
-    if let Err(e) = put.wait() {
-        tracing::error!("{}", e);
-        result::Z_EGENERIC
-    } else {
-        result::Z_OK
+    match put.wait() {
+        Ok(_) => result::Z_OK,
+        Err(e) if e.downcast_ref::<SessionClosedError>().is_some() => result::Z_SESSION_CLOSED,
```

Comment:
do we need some specific tracing message here and in other places ?

Reply:
Don't think so, because it's kind of included in the error kind. So the user can choose to ignore it without cluttering the logs.